src/lib.rs

```diff
@@ -169,16 +169,37 @@ impl Jieba {
         instance
     }
 
+    /// Create a new instance with dict
+    pub fn new_with_dict<R: BufRead>(dict: &mut R) -> Self {
```

Comment:
I'd prefer to name it `with_dict`. Also it might be better to return a `io::Result<Self>`.